client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDatabaseMetaDataWithBrokenCatalog.java

```diff
@@ -0,0 +1,789 @@
+import io.trino.spi.type.VarbinaryType;
+import io.trino.testing.CountingMockConnector;
+import io.trino.type.ColorType;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import static io.trino.jdbc.TestingJdbcUtils.array;
+import static io.trino.jdbc.TestingJdbcUtils.list;
+import static io.trino.jdbc.TestingJdbcUtils.readRows;
+import static io.trino.spi.type.CharType.createCharType;
+import static io.trino.spi.type.DecimalType.createDecimalType;
+import static io.trino.spi.type.TimeType.createTimeType;
+import static io.trino.spi.type.TimeWithTimeZoneType.createTimeWithTimeZoneType;
+import static io.trino.spi.type.TimestampType.createTimestampType;
+import static io.trino.spi.type.TimestampWithTimeZoneType.createTimestampWithTimeZoneType;
+import static io.trino.spi.type.VarcharType.createUnboundedVarcharType;
+import static io.trino.spi.type.VarcharType.createVarcharType;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(singleThreaded = true)
+public class TestTrinoDatabaseMetaDataWithBrokenCatalog
+{
+    private static final String TEST_CATALOG = "test_catalog";
+    private static final String COUNTING_CATALOG = "mock_catalog";
+
+    private CountingMockConnector countingMockConnector;
+    private TestingTrinoServer server;
+
+    private Connection connection;
+    private static final Properties PROPERTIES = new Properties();
+
+    static {
+        PROPERTIES.setProperty("user", "admin");
+        PROPERTIES.setProperty("sessionProperties", "ignore_broken_catalogs:true");
+    }
+
+    @BeforeClass
+    public void setupServer()
+            throws Exception
+    {
+        Logging.initialize();
+        server = TestingTrinoServer.create();
+
+        server.installPlugin(new TpchPlugin());
+        server.createCatalog(TEST_CATALOG, "tpch");
+
+        server.installPlugin(new BlackHolePlugin());
+        server.createCatalog("blackhole", "blackhole");
+
+        server.installPlugin(new HivePlugin());
+        server.createCatalog("hive", "hive", ImmutableMap.<String, String>builder()
+                // the Hive Metastore server doesn't exist resulting in Timeout Exceptions
+                .put("hive.metastore.uri", "thrift://localhost:9999")
+                .put("hive.security", "sql-standard")
```

Comment:
I was going to suggest adding `.put("hive.metastore-timeout", "10ms")` to fail faster, but it doesn't seem to be making much of a difference.

Reply:
I think that property is about the timeout for the hive metastore to respond after a successful request. In this case it is rather the connect timeout.